xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp

```diff
@@ -465,8 +465,18 @@ bool CDVDVideoCodecAndroidMediaCodec::Open(CDVDStreamInfo &hints, CDVDCodecOptio
         case FF_PROFILE_H264_HIGH_10:
           profile = CJNIMediaCodecInfoCodecProfileLevel::AVCProfileHigh10;
           break;
+        case FF_PROFILE_H264_HIGH_422:
+          profile = CJNIMediaCodecInfoCodecProfileLevel::AVCProfileHigh422;
+          break;
+        case FF_PROFILE_H264_HIGH_444:
+          profile = CJNIMediaCodecInfoCodecProfileLevel::AVCProfileHigh444;
+          break;
+        // All currently not supported formats
         case FF_PROFILE_H264_HIGH_10_INTRA:
-          // No known h/w decoder supporting Hi10P
+        case FF_PROFILE_H264_HIGH_422_INTRA:
+        case FF_PROFILE_H264_HIGH_444_PREDICTIVE:
+        case FF_PROFILE_H264_HIGH_444_INTRA:
+        case FF_PROFILE_H264_CAVLC_444:
           goto FAIL;
```

Comment:
should there be a  `default:` case here also?

Reply:
just to break?

Reply:
done!

Reply:
I was thinking more to fail for uncaught profiles, but I guess this wouldn't work for regular h264 profiles?

I'm not sure breaking here is what you want as it won't set the mime and formatname for profiles not captured in the switch/case.

Reply:
It's a switch in a switch. It will just continue after the profile switch, which it was done before. The profile is only used for checking before opening:
https://github.com/xbmc/xbmc/blob/774171d3bb8927f992758001e495d8315a1e09d4/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAndroidMediaCodec.cpp#L664 here

Standard AVC was already opened, without prior checking.